client/llb/state.go

```diff
@@ -48,48 +48,54 @@ func (s State) Value(k interface{}) interface{} {
 	return s.ctx.Value(k)
 }
 
-func (s State) Marshal() ([][]byte, error) {
-	list, err := marshal(s.Output().Vertex(), nil, map[digest.Digest]struct{}{}, map[Vertex]struct{}{})
+func (s State) Marshal() (Definition, error) {
```

Comment:
nit: I like when functions like this return pointers. As it contains slices/maps it shares the memory anyway.